cdapython/simple_parser.py

```diff
@@ -1,238 +1,355 @@
+    def nud(self, context: Parser) -> Query:
         """Compute the value of this token when between two expressions."""
         # Fetch the expression to the right, stopping at the next boundary
         # of same precedence
-        value = self
-        return str(value.text)
+        query = Query()
+        query.value = self.text
+
+        return query
 
 
 class ArrayType(Token):
 
     lbp = 10
 
-    def nud(self, context: Parser) -> str:
+    def nud(self, context: Parser) -> Query:
         """Compute the value of this token when between two expressions."""
         # Fetch the expression to the right, stopping at the next boundary
         # of same precedence
-        value = self
-        return str(value.text)
+        query = Query()
+        query.value = self.text
 
+        return query
 
-class Singlequotes(Token):
-    lbp = 10  # Precedence
 
-    def nud(self, context: Parser) -> str:
-        """Compute the value of this token when between two expressions."""
-        # Fetch the expression to the right, stopping at the next boundary
-        # of same precedence
-        value = self
-        return str(value.text)
+class Singlequotes(Doublequotes):
+    lbp = 10  # Precedence
 
 
 class IN(Token):
-    lbp = 4
+    lbp = 5
     query = Query()
 
-    def led(self, left: str, context: Parser) -> Query:
+    def led(self, left: Query, context: Parser) -> Query:
         """Compute the value of this token when between two expressions."""
         # Fetch the expression to the right, stopping at the next boundary
         # of same precedence
         right_side = context.expression(self.lbp)
         self.query.node_type = "IN"
-        self.query.l = left.strip()
-        self.query.r = right_side.strip()
-        return query
+        self.query.l = col(backwards_comp(left.value.strip()))
+        self.query.r = infer_quote(right_side.value.strip())
+        return self.query
 
 
 class LIKE(Token):
-    lbp = 4
+    lbp = 5
     query = Query()
 
-    def led(self, left: str, context: Parser) -> Query:
+    def led(self, left: Query, context: Parser) -> Query:
         """Compute the value of this token when between two expressions."""
         # Fetch the expression to the right, stopping at the next boundary
         # of same precedence
 
         right_side = context.expression(self.lbp)
         self.query.node_type = "LIKE"
-        self.query.l = left.strip()
-        self.query.r = right_side.strip()
-        return query
+        self.query.node_type, right_side = query_type_conversion(
+            self.query.node_type, right_side.value
+        )
+        self.query.l = col(backwards_comp(left.value.strip()))
+        self.query.r = infer_quote(right_side.value.strip())
+        return self.query
+
+
+class NULL(Token):
+    lbp = 4
+    query = Query()
+
+    def nud(self, context: Parser) -> Query:
+        right_side = context.expression(self.lbp)
+        self.query.node_type = "NOT"
```

Comment:
I think we want this to be unquoted?

Reply:
are you talking about the new type value?

Reply:
`node_type` is being passed as `NOT` here, I think we can either just remove this class altogether or change the node_type to `unquoted`